ipfs.go

```diff
@@ -176,30 +167,20 @@ func (p *Peer) setupDAGService() error {
 
 func (p *Peer) setupReprovider() error {
 	if p.cfg.Offline || p.cfg.ReprovideInterval < 0 {
-		p.reprovider = provider.NewOfflineProvider()
+		p.reprovider = provider.NewNoopProvider()
 		return nil
 	}
 
-	queue, err := queue.NewQueue(p.ctx, "repro", p.store)
+	prov, err := provider.New(p.store,
+		provider.DatastorePrefix(datastore.NewKey("repro")),
+		provider.Online(p.dht),
+		provider.ReproviderInterval(p.cfg.ReprovideInterval),
+		provider.KeyProvider(provider.NewBlockstoreProvider(p.bstore)))
 	if err != nil {
 		return err
 	}
+	p.reprovider = prov
```

Comment:
There were changes to the provider subsystem so this is what it should look like now